pkg/migrations/migrations_utils.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package migrations
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/clusterunique"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessionprotectedts"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/uuid"
+)
+
+type migrationsManagerImpl struct {
+	evalCtx   *eval.Context
+	pts       protectedts.Storage
+	txn       isql.Txn
+	sessionID clusterunique.ID
+}
+
+func GetMigrationsManager(
+	ctx context.Context,
+	evalCtx *eval.Context,
+	pts protectedts.Provider,
+	txn isql.Txn,
+	sessionID clusterunique.ID,
+) eval.MigrationsManager {
+	return migrationsManagerImpl{
+		evalCtx:   evalCtx,
+		pts:       pts.WithTxn(txn),
+		txn:       txn,
+		sessionID: sessionID,
+	}
+}
+
+// ProtectTableForSession lays a protected timestamp for the specified table
```

Comment:
> While I get that migrations are why we're exposing the builtin externally, inside the DB this feels like generic PTS+Session behavior, not something migration specific.

Yeah, I had a feeling that I'd get a comment like this.  I completely agree that this is not migration specific and feels more like generic PTS/Session behaviour.  That being said, the reason I created a new package was to explicitly take it out of the generic path.  As covered by your comments below, we don't want this to fall into general use.  By placing it in a separate package, and one named "migrations", my hope was that it would give someone pause if they want to use it for some other purpose internally.  I'd also be happy to reinforce that by adding a comment to the function that's a more explicit warning against reuse.